src/horust/healthcheck.rs

```diff
@@ -133,15 +138,66 @@ pub fn prepare_service(healthiness: &Healthiness) -> Result<(), std::io::Error>
     Ok(())
 }
 
+/* fn client_mock() -> bool {
+    loop {
+        match TcpStream::connect("127.0.0.1:9999") {
+            Ok(mut _stream) => {
+                println!("Successfully connected to server in port 9999");
+                return true;
+            }
+            Err(e) if e.kind() == ErrorKind::ConnectionRefused => {
+                continue;
+            }
+            Err(e) => {
+                println!("Failed to connect: {}", e);
+                return false;
+            }
+        }
+    }
+}
+ */
+
 #[cfg(test)]
 mod test {
     use crate::horust::error::Result;
     use crate::horust::formats::{Event, Healthiness, Service, ServiceName, ServiceStatus};
     use crate::horust::healthcheck;
     use crate::horust::healthcheck::healthchecks;
     use std::collections::HashMap;
+    use std::io::prelude::*;
+    use std::io::ErrorKind;
+    use std::net::TcpStream;
+    use std::net::{Ipv4Addr, SocketAddrV4, TcpListener};
+    use std::thread;
     use tempdir::TempDir;
-
+    fn start_server(socket: SocketAddrV4) {
+        let listener = TcpListener::bind(socket).unwrap();
+        listener
+            .set_nonblocking(true)
+            .expect("Cannot set to non-blocking");
+        // accept connections and process them, spawning a new thread for each one
+        println!("Server listening on port: {}", socket.port());
+        for stream in listener.incoming() {
+            match stream {
+                Ok(stream) => {
+                    // connection succeeded
+                    handle_test_connection(stream);
```

Comment:
Could you please close the webserver as soon as it has handled the request? a `return` or `break` here should suffice.